server/s3_put_object_action.cc

```diff
@@ -267,6 +268,10 @@ void S3PutObjectAction::fetch_object_info_success() {
   s3_log(S3_LOG_DEBUG, "", "%s Exit", __func__);
 }
 
+void S3PutObjectAction::set_null_object_version_id(std::string ver_id) {
+  null_object_version_id = std::move(ver_id);
```

Comment:
Which index table this new field will be added to? Object metadata or version list?

Reply:
This new field will added only object_list_index. See `S3ObjectMetadata::version_entry_to_json(bool obj_index)` function.
```
  if (obj_index) {
    root["version_key_null_in_index"] = null_object_version_id;
  }
```